tools/cxbe/Exe.cpp

```diff
@@ -130,10 +131,11 @@ Exe::Exe(const char *x_szFilename)
 
             uint32 raw_size = m_SectionHeader[v].m_sizeof_raw;
             uint32 raw_addr = m_SectionHeader[v].m_raw_addr;
+            uint32 virt_size = m_SectionHeader[v].m_virtual_size;
 
-            m_bzSection[v] = new uint08[raw_size];
+            m_bzSection[v] = new uint08[std::max(virt_size, raw_size)];
 
-            memset(m_bzSection[v], 0, raw_size);
+            memset(m_bzSection[v], 0, virt_size);
```

Comment:
Any reason not to use the actual allocated size when clearing (i.e., the result of the `max`)? At a glance it looks like a potential source of error, even if it's safe in practice.

Reply:
We'd just clear more bytes than necessary -  `virt_size > raw_size` will result in the buffer getting completely zeroed, `virt_size < raw_size` will result in the buffer being completely filled with a copy of what's in the file.

Reply:
Understood that it'd have no actual effect, it's more that it looks potentially erroneous so somebody trying to debug a similar issue to the one that led to this fix might waste time thinking that uninitialized section data was in play.

Reply:
Personally I'd prefer it to be like what the actual loader at runtime does, but I see what you mean. I've changed it, the performance difference is insignificant anyway.